test/Microsoft.AspNetCore.Razor.Test.Common/Language/Legacy/IntializeTestFileAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    public class IntializeTestFileAttribute : BeforeAfterTestAttribute
+    {
+        private static int _callCount = 0;
+        private static string _previousMethod;
+
+        public override void Before(MethodInfo methodUnderTest)
+        {
+            if (typeof(ParserTestBase).GetTypeInfo().IsAssignableFrom(methodUnderTest.DeclaringType.GetTypeInfo()))
+            {
+                var typeName = methodUnderTest.DeclaringType.Name;
+                ParserTestBase.FileName = $"TestFiles/ParserTests/{typeName}/{methodUnderTest.Name}";
+
+                if (methodUnderTest.GetCustomAttributes(typeof(TheoryAttribute), inherit: false).Length > 0)
+                {
+                    if (!string.IsNullOrEmpty(_previousMethod) &&
```

Comment:
There seems to be a race here causing this if to always be true. I'm looking in to it.